File: app/models/bill.py
from sqlalchemy import Enum
from app.database import Column, Model, SurrogatePK, db, reference_col
from app.models.enums import ApprovalStatus
from app.models.enums import BillStatus


class Bill(Model, SurrogatePK):
    __tablename__ = "bills"

    invoice_id = reference_col('invoices')
    invoice = db.relationship('Invoice', foreign_keys=[invoice_id], back_populates='bill')
    payment_method_id = Column(db.BigInteger, nullable=False)
    status = Column(Enum(BillStatus), nullable=False)
    approval_status = Column(Enum(ApprovalStatus), nullable=False)
    bill_cancellation_id = reference_col('bill_cancellations', nullable=True)
    bill_cancellation = db.relationship('BillCancellation', uselist=False, back_populates='bill')

File: backend/Invoicing/src/Invoicing.Domain/Entities/Bill.cs
using Invoicing.Domain.Common;

namespace Invoicing.Domain.Entities
{
    public class Bill : BaseEntity
    {
        public Invoice Invoice { get; set; }
        public long InvoiceId { get; set; }
        public long PaymentMethodId { get; set; }
        public BillStatus Status { get; set; }
        public Guid BillSecureUrl { get; set; } = Guid.NewGuid();
        public ApprovalStatus ApprovalStatus { get; set; }
        
        public long? BillCancellationId { get; set; }
        public BillCancellation BillCancellation { get; set; }
    }
}